shared/src/metabase/mbql/util.cljc

```diff
@@ -156,53 +156,60 @@
     [:is-empty field]  [:or  [:=  field nil] [:=  field ""]]
     [:not-empty field] [:and [:!= field nil] [:!= field ""]]))
 
+(defn- replace-field-or-expression
+  "Replace a field or expression inside :time-interval"
+  [m unit]
+  (mbql.match/replace m
+    [:field id-or-name opts] [:field id-or-name (assoc opts :temporal-unit unit)]
+    [:expression expression-name] [:expression expression-name]))
```

Comment:
I'm adding `[:expression expression-name]` for the sake of explicitly. We can just ignore it and it'll work just find tho.